src/blog/stylo-another-kind-of-rich-text-editor.md

```diff
@@ -0,0 +1,164 @@
+tags: "#javascript #showdev #webdev #webcomponents"
+image: "https://cdn-images-1.medium.com/max/1600/1*kPGH8RB1reGGiq6nU0UaIg.jpeg"
+canonical: "https://zhx6p-eqaaa-aaaai-abbrq-cai.raw.ic0.app/d/converting-svg-to-image-in-javascript"
+---
+
+![](https://cdn-images-1.medium.com/max/1600/1*kPGH8RB1reGGiq6nU0UaIg.jpeg)
+
+*Photo by [Amy Hirschi](https://unsplash.com/@amyhirschi?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText) on [Unsplash](https://unsplash.com/s/photos/pen?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText)*
+
+*****
+
+So, I developed a new open source WYSIWYG interactive editor for JavaScript. Its goal is to bring great user experience and interactivity to the web, for everyone, with no dependencies. I called it: [Stylo](https://stylojs.com/).
+
+*****
+
+### But Why?
+
+While developing last year the migration of our web editor for slides, [DeckDeckGo](https://deckdeckgo.com), to [DFINITY](https://dfinity.org)’s Internet Computer, we also initialized a new experimental concept to support documents as well.
+
+As a very first test, I wrote and published a blog post on our platform — [Converting SVG To Image (PNG, Webp, etc.) in JavaScript](https://zhx6p-eqaaa-aaaai-abbrq-cai.raw.ic0.app/d/converting-svg-to-image-in-javascript).
+
+Although I was satisfied with the result, I was of course also aware that further enhancements were needed.
+
+That’s why I had the idea to extract the rich text editor capabilities of the core of our application. By limiting the scope of these features, I assumed it would be easier to improve and fix what still needed to be done.
+
+In addition, by publishing the new library as a new open source repo, I was also hoping it would benefit from feedbacks and contributions.
+
+What can I add? I am an engineer, I like to code 😉.
+
+*****
+
+### Features
+
+Not looking to oversell what Stylo can do — or do differently — but here are in my humble opinion a list of cool things it can do:
+
+*****
+
+#### Interactive Design 🎯
+
+Unlike other WYSIWYG editor Stylo has no sticky toolbar. It is inspired by design and user experience of platforms that I like — i.e [Notion](https://www.notion.so/) and [Medium](https://medium.com/).
+
+![](https://cdn-images-1.medium.com/max/1600/1*D-YYgIa5cY0Ya8FAN0Lyiw.gif)
+
+When writing the document, user can use different types of paragraph that can be added through a floating menu. These elements can be simple HTML elements — i.e. h1, h2, code etc. — or more complex block — i.e. any Web Components.
+
+For styling purpose, another toolbar gives the user the ability to modify text — e.g. bold, italic, colors, etc. — and add hyperlinks.
+
+Last but not least, for an optimized experience on desktop, many actions can be performed without the need of the mouse. Opening the list of paragraphs can be triggered hitting “/” and **bold**, *italic* and `mark` are accessible through keyboard shortcuts.
```

Comment:
```suggestion
Last but not least, for an optimized experience on desktop, many actions can be performed without the mouse. Opening the list of paragraphs can be triggered hitting “/” and **bold**, *italic* and `mark` are accessible through keyboard shortcuts.
```

Reply:
note: I had never heard of `mark` until stylo

Reply:
should it be another element?